tutorials/transparent-proxy-and-filtering-on-k8s-with-initializers/index.md

```diff
@@ -0,0 +1,216 @@
+
+This tutorial uses the [tproxy-initializer](https://github.com/danisla/kubernetes-tproxy/tree/master/cmd/tproxy-initializer) Kubernetes Initializer to inject the sidecar InitContainer, ConfigMap and environment variables into a deployment when the annotation `"initializer.kubernetes.io/tproxy": "true"` is present. This tutorial also demonstrates how to deploy the tproxy Helm chart with the optional [Role Based Access Control](https://kubernetes.io/docs/admin/authorization/rbac/) support.
+
+Just like in the previous tutorial, the purpose of the [tproxy-sidecar](https://github.com/danisla/kubernetes-tproxy/tree/master/sidecar) container is to create firewall rules in the pod network to block egress traffic. The [tproxy-podwatch](https://github.com/danisla/kubernetes-tproxy/tree/master/cmd/tproxy-podwatch) controller watches for pod changes containing the annotation and automatically add/removes the local firewall `REDIRECT` rules to apply the transparent proxy to the pod.
+
+![architecture diagram](./tproxy_initializers_diagram.png)
+
+**Figure 1.** transparent proxy with initializers architecture diagram
+
+This tutorial also demonstrates how to use 
+
+## Objectives
+
+- Create a Kubernetes cluster with initializer and RBAC support using Google Container Engine
+- Deploy the tproxy, tproxy-initializer and the tproxy-podwatch pods using Helm
+- Deploy example apps with annotations to test external access to a Cloud Storage bucket
+
+## Before you begin
+
+This tutorial assumes you already have a GCP account and are familiar with the high level concepts of Kubernetes Pods and Deployments.
+
+## Costs
+
+This tutorial uses billable components of GCP, including:
+
+- [Google Container Engine](https://cloud.google.com/container-engine/pricing)
+
+Use the [Pricing Calculator](https://cloud.google.com/products/calculator/#id=f52c2651-4b02-4da3-b8cd-fdbca6ad89a9) to estimate the costs for your environment. The resources used in this tutorial are approximately $2.39 per day.
+
+## Checkout the source repository
+
+1. Open [Cloud Shell](https://console.cloud.google.com/cloudshell)
+
+2. Clone the repository containing the code for this tutorial:
+
+        git clone https://github.com/danisla/kubernetes-tproxy
+        cd kubernetes-tproxy
+
+    The remainder of this tutorial will be run from the root of the cloned repository directory.
+
+## Create Container Engine cluster and install Helm
+
+1. Create Container Engine cluster with alpha features enabled, RBAC support and a cluster version of at least 1.7 to support initializers:
```

Comment:
"Create a Container Engine cluster with alpha features enabled, RBAC support, and..."